crates/rust-analyzer/src/main_loop.rs

```diff
@@ -443,7 +451,22 @@ fn loop_turn(
         pool.execute({
             let subs = loop_state.subscriptions.subscriptions();
             let snap = world_state.snapshot();
-            move || snap.analysis().prime_caches(subs).unwrap_or_else(|_: Canceled| ())
+
+            let total = subs.len();
+
+            let mut progress = PrimeCachesProgressNotifier::begin(
+                connection.sender.clone(),
+                loop_state.next_request_id(),
+                total,
+            );
+
+            move || {
+                snap.analysis()
+                    .prime_caches(subs, move |i| {
+                        progress.report(i + 1);
```

Comment:
Note that we will initiate a new progress after every change, which is probably is not the right user experience. The `i` itself is also pretty misleading -- the number of open files (which is typically `1`) doesn't really have a relationship to the amount of work we need to do.